### ISSUE TITLE
API v1.1.1 returns unparsed string instead of object when querying list function

### ISSUE THREAD

**Issue by taris** [dominicbarnes/node-couchdb-api#9]
Hi!

I am querying a list function (http://dominicbarnes.us/node-couchdb-api/api/view/list.html) in my project and noticed that the behaviour changed from v1.0.0 to v1.1.1.
While the "response" in v1.0.0 returned an object, the "response" in v.1.1.1 contains the unparsed string.

``` javascript
1.0.0: Value: [object Object] - Typeof: object
1.1.1: Value: {"total":19790} - Typeof: string
```

Use this ddoc for testing:

``` javascript
{
   "_id": "_design/list-test",
   "language": "javascript",
   "views": {
       "view-func": {
           "map": "function(oDocument)\n{\nemit(oDocument._id, 1)\n}"
       }
   },
   "lists": {
       "list-func": "function(oHead, oViewRequest)\n{\n    var iTotalSum = 0;\n    while (oCurrRow = getRow())\n    {\n        iTotalSum += oCurrRow.value;\n    }\n\n    send(toJSON({total: iTotalSum}));\n}"
   }
}
```

and this script for executing with node:

``` javascript
var couchdb = require("couchdb-api");

/** default values: "localhost", 5984 **/
var server = couchdb.srv("localhost", 5984);

var db = server.db("database");

db.ddoc("list-test").view("view-func").list("list-func", function (err, response) {
    console.log("Value: "+response+" - Typeof: "+(typeof response));
});
```

Is this a bug or do i need to call it differently? The documentation says it should be an object (http://dominicbarnes.us/node-couchdb-api/api/callbacks.html)

Thx!

**Comment by dominicbarnes** [comment]
Sorry, my documentation hasn't quite caught up to some of the more subtle API changes I've made in this release. The signature for list now looks like `function (list, query, options, callback)` (the new param is #3, `options`)

I needed to ensure that I didn't assume all responses to be JSON, especially list functions that may or may not output other content types. To force the body to be parsed as JSON, pass `{ json: true }` as the options param. Note, you'll also need to explicitly pass `null` as `query`.

```
view.list("list-func", null, { json: true }, function (err, response) {
    console.log("Value: "+response+" - Typeof: "+(typeof response));
});
```

Sorry, but I've made your code a little bit more verbose. I'll try to have my documentation updated today.
